File: services/api_gateway/types/response/channel.go
package response

type ChannelResponse struct {
	ID              string       `json:"id"`
	OwnerID         string       `json:"owner_id"`
	Name            string       `json:"name"`
	Handle          string       `json:"handle"`
	Description     string       `json:"description"`
	BannerURL       string       `json:"banner_url"`
	SubscriberCount int32        `json:"subscriber_count"`
	TotalViews      int64        `json:"total_views"`
	TotalVideos     int32        `json:"total_videos"`
	CreatedAt       string       `json:"created_at"`
	UpdatedAt       string       `json:"updated_at"`
	Owner           UserResponse `json:"owner"`
}

File: web/types/channel.ts
export type ChannelResponse = {
  id: string;
  owner_id: string;
  name: string;
  handle: string;
  description: string;
  banner_url: string;
  subscriber_count: number;
  total_views: number;
  total_videos: number;
  created_at: string;
  updated_at: string;
  owner: OwnerResponse;
};

type OwnerResponse = {
  id: string;
  email: string;
  name: string;
  picture: string;
  created_at: string;
  updated_at: string;
};